lib/msprime.c

```diff
@@ -807,53 +807,123 @@ msp_verify_segments(msp_t *self, bool verify_breakpoints)
     }
 }
 
-/* FIXME logic doesn't make sense in continuous space */
+/* Find the segment that spans the given position */
```

Comment:
Same logic can apply in the c code here. We can make a 'local' segment_finder_t struct (or "class") which would do the same thing as the Python one. It'll simplify things a bit for someone reading the code if this functionality is grouped together and not part of the ``msp_`` namespace (which is poorly named btw, should be something like ``msp_sim_t``, but that's another story)

Reply:
Yes this is much cleaner.